docs/design/proposed/interactive-mode.md

```diff
@@ -0,0 +1,47 @@
+
+## Goals
+
+* Provide a guided experience for users to provide values for symbols (and other items like the project name)
+* Enforce required symbols, but allow for skipping optional values
+* Support conditional symbols
+* Only provide the interactive experience when a user has entered a command (i.e. isatty is true)
+* Where possible, provide an attractive and informative prompting experience through the use of ANSI control codes and terminal formatting. When the environment doesn't support it, use 'boring' formatting.
+
+## Non-Goals
+
+* Provide interactivity to the entire .NET CLI
+* Extend interactivity to commands like `dotnet new search` or `dotnet new list` (i.e. interactive filtering)
+
+## Workflow
+
+When a user invokes a template, the command should evaluate the user-provided parameters and determine if it should enter interactive mode. This is done by checking if the command is being invoked from a terminal (i.e. isatty is true) and if the command has any symbols that are of type `parameter` and are required _while also having required symbols that are not explicitly provided_. If both of these conditions are true, the command should enter interactive mode.
+
+When interactive mode is entered, the command should prompt the user for each required symbol. If the symbol has a default value, it should be displayed as the default value with some formatting (for example, grayed out text). If the symbol has a description, it should be displayed as part of a prompt. If the symbol has a datatype, the command should validate the input and prompt the user again if the input is invalid. At each stage of the prompt, the user should be able to press a key to skip the prompt and use the default value (if one exists). At each stage of the prompt, the user should be able to press a key to skip all other prompts and use their default values (if they exist). Finally, at each stage of the prompt the user should be able to press a key and quit the entire operation.
+
+### Concerns/open questions
+
+* Many templates have symbols that are 'internal' and/or easily confused - e.g. TargetFrameworkOverride and Framework. How to coalesce these to prevent multiple prompts for something a user views as one item?
+* Some parameters are not something I'd expect a user to ever want to change - e.g. skipRestore. Should we have another property marker that hides them?
+  * For initial version, prompt for these too. We can tweak logic based on feedback.
+* Which of the 'general' invocation options like -n and -o should we present here?
+* Should we have a new `prompt` property for this? Descriptions feel odd grammatically (which is why I didn't use them in the below example).
```

Comment:
this belongs to `dotnetcli.host.json` imo

Reply:
That's a good point - if the specific data becomes more general then we could promote it to the template config.